kong.go

```diff
@@ -219,7 +219,10 @@ func (k *Kong) Parse(args []string) (ctx *Context, err error) {
 		return nil, &ParseError{error: err, Context: ctx}
 	}
 	if err = ctx.Validate(); err != nil {
-		return nil, &ParseError{error: err, Context: ctx}
+		if !k.usageOnError || !isMissingChildError(err) {
+			return nil, &ParseError{error: err, Context: ctx}
+		}
+		ctx.Error = err
```

Comment:
@alecthomas after opening this I am thinking that it actually makes more sense to perform this check in the global `Parse`. However, that would require returning the `ctx` and the `ParseError` together. Thoughts?

Reply:
I think this is fine for now.